File: Excel/500-599/581/581 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/581 Increment Sequences.xlsx"
input = read_excel(path, range = "A1:A18")
test  = read_excel(path, range = "B1:B18")

result = input %>% 
  mutate(Answer_Expected = if_else(is.na(Column1), 0, cumsum(!is.na(Column1) & (is.na(lag(Column1)) | lag(Column1) != Column1))))

all.equal(result$Answer_Expected, test$`Answer Expected`)
#> [1] TRUE

File: Excel/500-599/581/581 Challenge.py
import pandas as pd
import numpy as np

path = "581 Increment Sequences.xlsx"
input = pd.read_excel(path, usecols="A", nrows=18)
test = pd.read_excel(path, usecols="B", nrows=18)

input['Answer_Expected'] = np.where(
    input['Column1'].isna(), 
    0, 
    (input['Column1'].notna() & (input['Column1'] != input['Column1'].shift(1).fillna(''))).cumsum()
).astype('int64')

print(input['Answer_Expected'].equals(test['Answer Expected']))

print(input.dtypes)
print(test.dtypes)